synapse/storage/databases/main/roommember.py

```diff
@@ -697,6 +697,21 @@ async def get_rooms_for_users(
 
         return {key: frozenset(rooms) for key, rooms in user_rooms.items()}
 
+    async def get_rooms_for_users(
+        self, user_ids: Collection[str]
+    ) -> Dict[str, FrozenSet[str]]:
+        """A batched wrapper around `_get_rooms_for_users`, to prevent locking
```

Comment:
`_get_rooms_for_users` is itself a batched version of something else. It's batching all the way down! :-)